### PR TITLE
add missing modules for students and auth,

### PR DESCRIPTION
**Pull request by RadekJ87**
add missing modules for students and auth,
new packages @nestjs/passport passport passport-jwt cookie-parser
created jwt strategy

### CODE REVIEW

File: src/auth/auth.controller.ts [hunk 1]
```diff
@@ -1,4 +1,26 @@
...
+import { UserObj } from '../decorators/portal-users.decorator';
+import { Student } from '../student/student.entity';
 
 @Controller('auth')
-export class AuthController {}
+export class AuthController {
+   constructor(private readonly authService: AuthService) {}
+
+   @Post('/login')
+   async phoneRegister(
```
Comment (Fremen1990) [inline src/auth/auth.controller.ts]: this function will not be named 'phoneRegister'  :)  call it just 'login' 👍

File: src/auth/auth.service.ts [hunk 2]
```diff
@@ -1,4 +1,82 @@
...
+               //domena ciastka
+               domain: 'localhost',
+               //FE nie widzi ciastka
+               httpOnly: true,
+            })
+            .json({ ok: true });
+      } catch (e) {
+         return res.json({ error: e.message });
+      }
+   }
```
Comment (Fremen1990) [inline src/auth/auth.service.ts:66]: Controller is called 'phoneRegister' and servie 'login'

File: src/auth/jwt.strategy.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+   async validate(payload: JwtPayload, done: (error, user) => void) {
+      if (!payload || !payload.id) {
+         return done(new UnauthorizedException(), false);
+      }
+
+      // sprawdzanie typu uzytkownika, dodatkowy parametr w metodzie i endpoincie np. login/student, login/hr?
+      // Poki co dla student
+      const user = await Student.findOneBy({
+         currentSessionTokenId: payload.id,
+      });
```
Comment (Fremen1990) [inline src/auth/jwt.strategy.ts:37]: Here we will take User table and there will be role used to redirect user to proper validated routes

**Review by Fremen1990 — CHANGES_REQUESTED** [review]

File: src/auth/auth.controller.ts (hunk 1 above)
Comment (RadekJ87) [reply src/auth/auth.controller.ts]: fixed in newest commit ;)

**Review by Fremen1990 — APPROVED** [review]
👍

**Pull request merged**